e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts

```diff
@@ -0,0 +1,288 @@
+    const complianceNodeId = uuidv4();
+    const clientRunsNodeId = uuidv4();
+    const reportId = uuidv4();
+    const nodeName = `${cypressPrefix}-${Cypress.moment().format('MMDDYYhhmm')}`;
+    const start = Cypress.moment().utc().subtract(3, 'day').startOf('day').format();
+    const end = Cypress.moment().utc().endOf('day').format();
+
+    // Ingest a InSpec report with attribues that match all the projects
+    cy.fixture('compliance/inspec-report.json').then((report) => {
+      report.organization_name = '75th Rangers';
+      report.source_fqdn = 'example.org';
+      report.environment = 'arctic';
+      report.policy_group = 'red_ring';
+      report.policy_name = 'fire';
+      report.roles = ['backend'];
+      report.chef_tags = ['v3'];
+      report.node_uuid = complianceNodeId;
+      report.node_name = nodeName;
+      report.end_time = Cypress.moment().utc().format();
+      report.report_uuid = reportId;
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: report
+      });
+    });
+
+    // Ingest a node with attribues that match all the projects
+    cy.fixture('converge/avengers1.json').then((node) => {
+      node.organization_name = '75th Rangers';
+      node.chef_server_fqdn = 'example.org';
+      node.node.chef_environment = 'arctic';
+      node.policy_group = 'red_ring';
+      node.policy_name = 'fire';
+      node.node.automatic.roles = ['backend'];
+      node.node.normal.tags = ['v3'];
+      node.entity_uuid = clientRunsNodeId;
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: node
+      });
+    });
+
+    // wait for the report to be ingested
+    waitForNodemanagerNode(complianceNodeId, 30);
+    // wait for the client run report to be ingested
+    waitForNodemanagerNode(clientRunsNodeId, 30);
+
+    // Ensure the both the compliance and client run nodes' tags are updated
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: {
+          projects: [projectWithRule.project.id],
+          'api-token': Cypress.env('ADMIN_TOKEN')
+        },
+        method: 'POST',
+        url: '/api/v0/nodes/search',
+        body: {
+          order: 'DESC',
+          page: 1,
+          per_page: 100
+        }
+      }).then((response) => {
+        expect(response.body.nodes.length).to.greaterThan(0);
+        expect(nodeExist(complianceNodeId, response.body.nodes)).to.equal(true);
+        expect(nodeExist(clientRunsNodeId, response.body.nodes)).to.equal(true);
+      });
+    });
+  });
+});
+
+function nodeExist(nodeId: string, nodes: any[]): boolean {
+  for (const node of nodes ) {
+    if (node.id === nodeId) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function waitForNodemanagerNode(nodeId: string, maxRetries: number) {
+  cy.request({
```

Comment:
starting to think we should have a helper that makes the request with the token 🤔

Reply:
That would be helpful 😄